extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/controller/jsonrpc.js

```diff
@@ -126,7 +126,7 @@ export class JsonRpc {
                 JsonRpc.serverUri = "ws:";
             }
             var currentPath = window.location.pathname;
-            currentPath = currentPath.substring(0, currentPath.indexOf('/dev')) + "/dev-ui";
+            currentPath = currentPath.substring(0, currentPath.indexOf('/dev-ui')) + "/dev-ui";
```

Comment:
I think to be completely safe, it would have to be refined a bit:
- you would have to forbid `dev-ui` as a root path
- or be sure you would have `/dev-ui` after the first iteration and use `lastIndexOf` - but if you have additional path segments, you can't really be sure

Also, I think you should:
- check if the string ends with `/dev-ui` and do what you're doing
- if not, I would search for `/dev-ui/` instead.

Otherwise, you improved the situation for sure but you can still break it relatively easily if you're unlucky (root path of `dev-ui-whatever` for instance).

Reply:
I applied you suggestions. I tested with setting the root to both `/dev-ui` and `/dev-ui-whatever`.